In [13]:
import numpy as np
import pandas as pd
from scipy import stats
from scipy.stats import describe
from dataclasses import asdict

In [14]:
df=pd.read_csv(r"C:\Users\sanum\Desktop\assignment\assignments data\hypothesis testing\Cutlets.csv")
df.head()

,Unit A,Unit B
0,6.8090,6.7703
1,6.4376,7.5093
2,6.9157,6.7300
3,7.3012,6.7878
4,7.4488,7.1522


In [18]:
df.rename(columns={'Unit A':'A' ,'Unit B':'B'},inplace=True)
D=describe(df)
df.head()

,A,B
0,6.8090,6.7703
1,6.4376,7.5093
2,6.9157,6.7300
3,7.3012,6.7878
4,7.4488,7.1522


In [19]:
varA=df['A'].var()
varB=df['B'].var()

In [20]:
print("Inferences from the data")
for key,value in D._asdict().items():
    print(f"{key}:{value}")
Variance=[varA,varB]
print("variance:", Variance)

Inferences from the data
nobs:35
minmax:(array([6.4376, 6.038 ]), array([7.5169, 7.5459]))
mean:[7.01909143 6.96429714]
variance:[0.08317945 0.117924  ]
skewness:[-0.11831877 -0.36346397]
kurtosis:[-0.96159273  0.00599388]
variance: [0.08317945374789915, 0.11792400440336127]


# Checking Normality of the data

Given confidence level = 95%

α = 0.05 (1-0.95)

H0 = Both Unit A and Unit B are normal.

H1 = Unit A and Unit B are not normal.

In [21]:
stats.shapiro(df["A"])

ShapiroResult(statistic=0.9649459719657898, pvalue=0.31998491287231445)

In [22]:
stats.shapiro(df["B"])

ShapiroResult(statistic=0.9727305769920349, pvalue=0.5225146412849426)

By observing the p-values of both the UnitA and UnitB we can observe that the

Both the p-values > α. i.e., (0.320>0.05) and (0.522>0.05)

So, we rejcet the alternate hypothesis(H1) and accept the Null hypothesis(H0).

Therefore, The Unit-A and Unit-B are normally distributed.

# Performing Paired T-Test

Significance level = 0.05

H0 == Mean for Y1 and Y2 are equal

H1 == Mean for Y1 and Y2 are not equal


As there is no population information, we cannot find the Z - Score for the given information
So we will calculate the T-Test for the cutlets.csv

In [23]:
stats.ttest_rel(df['A'],df['B'])

TtestResult(statistic=0.7536787225614316, pvalue=0.4562300768038412, df=34)

By performing a paired t-test, we got a p-value of 0.45, which is greater than the significance level (α = 0.05).
Since p-value > α, we fail to reject the null hypothesis.

Conclusion: There is no significant difference between the average cutlet weights of Unit A and Unit B.